pkg/gatewayserver/gatewayserver.go

```diff
@@ -972,6 +990,36 @@ func (gs *GatewayServer) handleLocationUpdates(ctx context.Context, conn connect
 	}
 }
 
+// handleVersionInfoUpdates updates gateway attributes with version info.
+// This function runs exactly once; only for the first status message of each connection, since version information does (should) not change within the same connection.
+func (gs *GatewayServer) handleVersionInfoUpdates(ctx context.Context, conn connectionEntry) {
+	select {
+	case <-ctx.Done():
+		return
+	case <-conn.VersionInfoChanged():
+		status, _, ok := conn.StatusStats()
+		if ok && status.Versions["model"] != "" && status.Versions["firmware"] != "" {
```

Comment:
Add alias for `status.Versions` and `conn.Gateway().Attributes`.

If `conn.Gateway().Attributes` is `nil`, this panics.
Also make sure that `status.Versions` is not `nil` (I'm not sure if we initialize this everywhere).

Reply:
I don't think so. As far as I'm aware, you can read from an uninitialised map.
https://go.dev/play/p/TGaXMKqyTXc

Reply:
Correct, only write panics on `nil` maps.